server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -358,6 +359,53 @@ public TabletServer(ServerConfigurationFactory confFactory, VolumeManager fs) th
     this.logSorter = new LogSorter(instance, fs, aconf);
     this.replWorker = new ReplicationWorker(this, fs);
     this.statsKeeper = new TabletStatsKeeper();
+    final int numTopTabletsToLog = aconf.getCount(Property.TSERV_LOG_TOP_TABLETS_COUNT);
+    final long logTopTabletsDelay = aconf.getTimeInMillis(Property.TSERV_LOG_TOP_TABLETS_INTERVAL);
+    if (numTopTabletsToLog > 0) {
+      SimpleTimer.getInstance(aconf).schedule(new Runnable() {
+        @Override
+        public void run() {
+          Comparator<Pair<String,Long>> topTabletComparator = new Comparator<Pair<String,Long>>() {
+            @Override
+            public int compare(Pair<String,Long> first, Pair<String,Long> second) {
+              return second.getSecond().compareTo(first.getSecond());
+            }
+          };
+          PriorityQueue<Pair<String,Long>> topTabletsByIngestCount =
+              new PriorityQueue<>(numTopTabletsToLog, topTabletComparator);
+          PriorityQueue<Pair<String,Long>> topTabletsByQueryCount =
+              new PriorityQueue<>(numTopTabletsToLog, topTabletComparator);
+          synchronized (onlineTablets) {
+            for (Tablet tablet : onlineTablets.values()) {
+              addToTopTablets(tablet.totalIngest(), topTabletsByIngestCount, numTopTabletsToLog);
+              addToTopTablets(tablet.totalQueries(), topTabletsByQueryCount, numTopTabletsToLog);
+            }
+            logTopTablets(topTabletsByIngestCount, "QUERY", numTopTabletsToLog);
+            logTopTablets(topTabletsByQueryCount, "INGEST", numTopTabletsToLog);
+          }
+        }
+
+        private void addToTopTablets(long count,
+            PriorityQueue<Pair<String,Long>> topTabletsQueue, int numTopTabletsToLog) {
+          if (topTabletsQueue.size() < numTopTabletsToLog
+              || topTabletsQueue.peek().getSecond() < count) {
+            if (topTabletsQueue.size() == numTopTabletsToLog) {
+              topTabletsQueue.remove();
+            }
+          }
+        }
+
+        private void logTopTablets(PriorityQueue<Pair<String,Long>> topTabletsQueue,
+            String label, int numTopTabletsToLog) {
+          for (int i = 0; i < numTopTabletsToLog; i++) {
```

Comment:
I am not sure what you are logging... what do you mean by "top"?

Reply:
ill refactor to make it more clear